mongodbatlas/provider.go

```diff
@@ -54,86 +56,88 @@ func Provider() *schema.Provider {
 		Schema: map[string]*schema.Schema{
 			"public_key": {
 				Type:     schema.TypeString,
-				Required: true,
```

Comment:
We need to make sure that the config from both provider matches exactly. Hence, we need to make some changes to the existing SDKv2 provider so that the schema is same.
Moved the `DefaultFunc` logic to configure method.
See https://discuss.hashicorp.com/t/muxing-upgraded-tfsdk-and-framework-provider-with-default-provider-configuration/43945/4